src/global/variables/clauses/IntegerLiteral.java

```diff
@@ -1,26 +1,67 @@
 package global.variables.clauses;
 
+import global.utils.RegexUtil;
 import global.variables.Clause;
 
 public class IntegerLiteral extends Clause {
     int value;
+    int upper;
+    int lower;
 
     public IntegerLiteral(int value) {
         super();
         this.value = value;
+        this.upper = value;
+        this.lower = value;
+
     }
 
     public IntegerLiteral(int value, String name) {
         super(name);
         this.value = value;
+        this.upper = value;
+        this.lower = value;
+    }
+
+    public IntegerLiteral(int lower, int upper) {
+        super();
+        this.value = 0;
```

Comment:
Why set this value at all here?

Reply:
Fair point. I can probably just have a default

Reply:
I think just leave it null though, if the literal has a range then it doesn't need to store a value for any reason, right?